src/llama_recipes/finetuning.py

```diff
@@ -228,12 +238,29 @@ def main(**kwargs):
             use_kahan_summation=False,
             weight_decay=train_config.weight_decay,
         )
+    elif optimizer_in_backward_available:
+        print(f"setting up optimizer overlap")
+        optim_kwargs = {"lr": train_config.lr}
```

Comment:
Please see if setting foreach=False param for the optimizer will bring further memory savings

Reply:
@chauhang foreach may help issue less CUDA kernel calls overall and reduce kernel launch overhead, but I don't see how it would result in more memory savings.